File: src/Proveedor/domain/Proveedor.ts
import type { ProveedorId } from "./ProveedorId";
import type { ProveedorNombre } from "./ProveedorNombre";
import type { ProveedorDireccion } from "./ProveedorDireccion";
import type { ProveedorTelefono } from "./ProveedorTelefono";
import type { ProveedorFechaCreacion } from "./ProveedorFechaCreacion";
import type { ProveedorFechaModificacion } from "./ProveedorFechaModificacion";
import type { ProveedorEsActivo } from "./ProveedorEsActivo";

export class Proveedor {
    id: ProveedorId;
    nombre: ProveedorNombre;
    direccion: ProveedorDireccion;
    telefono: ProveedorTelefono;
    fechaCreacion: ProveedorFechaCreacion;
    fechaModificacion: ProveedorFechaModificacion;
    esActivo: ProveedorEsActivo;

    constructor(
        id: ProveedorId,
        nombre: ProveedorNombre,
        direccion: ProveedorDireccion,
        telefono: ProveedorTelefono,
        fechaCreacion: ProveedorFechaCreacion,
        fechaModificacion: ProveedorFechaModificacion,
        esActivo: ProveedorEsActivo
    ) {
        this.id = id;
        this.nombre = nombre;
        this.direccion = direccion;
        this.telefono = telefono;
        this.fechaCreacion = fechaCreacion;
        this.fechaModificacion = fechaModificacion;
        this.esActivo = esActivo;
    }
}

File: desktop_app/src/main/java/com/model/Proveedor.java
package com.model;

import java.util.Date;

public class Proveedor {
    private int id;
    private String nombre;
    private String direccion;
    private String telefono;
    private Date fechaCreacion;
    private Date fechaModificacion;

    public Proveedor() {}

    public Proveedor(
            int id,
            String nombre,
            String direccion,
            String telefono,
            Date fechaCreacion,
            Date fechaModificacion
    ) {
        this.id = id;
        this.nombre = nombre;
        this.direccion = direccion;
        this.telefono = telefono;
        this.fechaCreacion = fechaCreacion;
        this.fechaModificacion = fechaModificacion;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public Date getFechaCreacion() {
        return fechaCreacion;
    }

    public void setFechaCreacion(Date fechaCreacion) {
        this.fechaCreacion = fechaCreacion;
    }

    public Date getFechaModificacion() {
        return fechaModificacion;
    }

    public void setFechaModificacion(Date fechaModificacion) {
        this.fechaModificacion = fechaModificacion;
    }
}
